### PR TITLE
Universion

### PR DESCRIPTION
**Pull request by yphus**
## Description

Each sub-project of the Checkbox stack (`checkbox-ng`, `checkbox-support`, the different `providers`, etc.) have their own version number. This is because, historically, each of these is distributed as its own Debian package, and each package lives independently from the others.

In parallel, Checkbox snaps have also their own version number (at time of writing, checkbox is `v2.3` and checkbox[16..22] is `v1.31`)

The multiplication of version numbers brings confusion.

Since we always release the whole stack at once, and since we plan to provide more regular releases coming on, it makes more sense to version everything with a single version number.

This means that some sub-projects  that did not receive any updates (for instance the TPM2 provider that rarely gets updated nowadays) would see their version bumped, but it would be easier to track the changes and the releases this way, while also bringing clarity to users.

This PR first aligns all versions to v2.3, for all sub projects in the monorepo.
A single `.bumpversion.cfg` file is provided at the root of the repository.
A new "Bumpversion" manual worklow is available to bump the stack version (minor increment by default, major being the second choice).
This workfklow opens a PR to review and merge with instruction to tag the release.

A single tag pushed to origin will then trigger 3 workflows and open a draft release (e.g https://github.com/canonical/checkbox-staging/releases/tag/untagged-45c1db99e181092369c3):
- debs
- checkbox snaps
- checkbox core snaps

Finally the deb release workflow is greatly simplified, no intermediate RC tags.
Binary packages built in the testing ppa already have the stable version, meaning there won't be a rebuild in stable if validation tests are ok, only copies.

_Note: The promotion/copies from beta to stable is not covered by the PR (new workflows are required to copy deb packages and promote snaps automatically)_

## Resolved issues

- https://warthogs.atlassian.net/browse/CHECKBOX-461
- https://warthogs.atlassian.net/browse/CHECKBOX-486
- https://warthogs.atlassian.net/browse/CHECKBOX-487

## Documentation

- WIP, will be provided as a separate PR once a new beta release is out.

## Tests

- [ ] Steps to follow for reviewer to run & manually test are provided.
- [X] A list of what tests were run and on what platform/configuration is provided.

Tested using the https://github.com/canonical/checkbox-staging repository.
- Creation of a new version (2.4) for the whole stack using the bumpversion workflow
- A new PR is automatically created with instruction to tag the release and push it to origin.
- the new tag triggers all the desired workflows (debs and snaps)
- a draft release is created
- beta snaps are built using snapcraft remote build as before
- beta debs are built using a new versioning (w/o Release candidate suffix) to copy binaries to stable PPA if validation tests are ok

### CODE REVIEW

File: tools/release/bumpversion.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+shopt -s globstar
+git config --global user.email "robot@canonical.com"
+git config --global user.name "Devices Certification Bot"
+git checkout -b bumpversion
+pushd $(git rev-parse --show-toplevel) > /dev/null
+cur_version=$(bumpversion --allow-dirty --dry-run --list ${1-minor} | grep current_version= | sed -r s,"^.*=",,)
```
Comment (kissiel) [inline tools/release/bumpversion.sh:8]: I'm guessing the double 'minor' default is on purpose (the first incarnation being in the workflow definition).

File: checkbox-ng/checkbox_ng/__init__.py [hunk 2]
```diff
@@ -24,4 +24,5 @@
 CheckBoxNG is a new version of CheckBox built on top of PlainBox
```
[the remark this answers is not in the document]
Comment (kissiel) [reply checkbox-ng/checkbox_ng/__init__.py:24]: btw. I know this is not a part of this patch, just got included as context to a patch here

**Review by kissiel — APPROVED** [review]
Awesome stuff. Looks and works clean (bumped locally).
One double question below.

File: .bumpversion.cfg [hunk 3]
```diff
@@ -0,0 +1,81 @@
+[bumpversion]
+current_version = 2.3
+parse = (?P<major>\d+)\.(?P<minor>\d+)
+serialize =
```
[the remark this answers is not in the document]
Comment (yphus) [reply .bumpversion.cfg]: Good catch, fixed

File: tools/release/bumpversion.sh (hunk 1 above)
Comment (yphus) [reply tools/release/bumpversion.sh:8]: double protection, but harmless.

**Pull request merged**